commit fc6097d4fb2988f12954993be70b882a345a35e5
Author: Anshuman Gupta <anshuman.gupta@intel.com>
Date:   Mon Jan 11 13:41:06 2021 +0530

    drm/i915/hdcp: DP MST transcoder for link and stream
    
    Gen12 has H/W delta with respect to HDCP{1.x,2.x} display engine
    instances lies in Transcoder instead of DDI as in Gen11.
    
    This requires hdcp driver to use mst_master_transcoder for link
    authentication and stream transcoder for stream encryption
    separately.
    
    This will be used for both HDCP 1.4 and HDCP 2.2 over DP MST
    on Gen12.
    
    Cc: Ramalingam C <ramalingam.c@intel.com>
    Reviewed-by: Uma Shankar <uma.shankar@intel.com>
    Reviewed-by: Ramalingam C <ramalingam.c@intel.com>
    Tested-by: Karthik B S <karthik.b.s@intel.com>
    Signed-off-by: Anshuman Gupta <anshuman.gupta@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20210111081120.28417-6-anshuman.gupta@intel.com

diff --git a/drivers/gpu/drm/i915/display/intel_hdcp.h b/drivers/gpu/drm/i915/display/intel_hdcp.h
index 1bbf5b67ed0a..bc51c1e9b481 100644
--- a/drivers/gpu/drm/i915/display/intel_hdcp.h
+++ b/drivers/gpu/drm/i915/display/intel_hdcp.h
@@ -25,7 +25,7 @@ void intel_hdcp_atomic_check(struct drm_connector *connector,
 int intel_hdcp_init(struct intel_connector *connector, enum port port,
 		    const struct intel_hdcp_shim *hdcp_shim);
 int intel_hdcp_enable(struct intel_connector *connector,
-		      enum transcoder cpu_transcoder, u8 content_type);
+		      const struct intel_crtc_state *pipe_config, u8 content_type);
 int intel_hdcp_disable(struct intel_connector *connector);
 void intel_hdcp_update_pipe(struct intel_atomic_state *state,
 			    struct intel_encoder *encoder,